pkg/logs/input/docker/scanner.go

```diff
@@ -33,39 +32,53 @@ type Scanner struct {
 	tailers          map[string]*Tailer
 	cli              *client.Client
 	registry         auditor.Registry
-	isRunning        bool
 	stop             chan struct{}
 }
 
 // NewScanner returns an initialized Scanner
-func NewScanner(sources *config.LogSources, pipelineProvider pipeline.Provider, registry auditor.Registry) *Scanner {
-	return &Scanner{
+func NewScanner(sources *config.LogSources, pipelineProvider pipeline.Provider, registry auditor.Registry) (*Scanner, error) {
+	scanner := &Scanner{
 		pipelineProvider: pipelineProvider,
 		sources:          sources,
 		tailers:          make(map[string]*Tailer),
 		registry:         registry,
 		stop:             make(chan struct{}),
 	}
+	err := scanner.setup()
+	if err != nil {
+		scanner.reportErrorToAllSources(err)
```

Comment:
would we have the edge case where we use the kubernetes integration without docker, and it'd be expected for docker to fail but we would still report the error in the agent status?

Reply:
I think this should not happen as we would not have any docker configuration.